e2e_tests/tests/all_providers/multitenancy.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use e2e_tests::TestClient;
+use parsec_client::core::interface::requests::{ProviderID, ResponseStatus};
+
+// These tests are executed by different users in the following order:
+// 1. client1_before is executed as parsec-client-1
+// 2. client2 is executed as parsec-client-2
+// 3. client1_after is executed as parsec-client-1
+//
+// They are executed against all possible authenticators in Parsec.
+
+#[test]
+fn client1_before() {
+    // Create one key on each provider
+    let mut client = TestClient::new();
+    client.do_not_destroy_keys();
+    client.set_direct_auth("client1".to_string());
```

Comment:
I'm not sure I get the purpose of these tests - shouldn't they use peer credential auth?

Reply:
They do use Peer Credentials as well! This line will get ignored if the default authentication selected in the client during construction is `UnixPeerCredentials`. This directly call `set_default_auth(Some("client1"))` and will do something when direct authentication is used.
This is done so that the same test can be used for direct authenticaiton and peer cred/jwt.

Reply:
Hmm, then maybe the method name is a bit misleading - why not just use the same name, `set_default_auth`?

Reply:
Changed it to the same name!